mailpoet/lib/Segments/DynamicSegments/Filters/WooCommerceNumberOfOrders.php

```diff
@@ -44,31 +44,55 @@ public function apply(QueryBuilder $queryBuilder, DynamicSegmentFilterEntity $fi
 
     $date = Carbon::now()->subDays($days);
 
-    $queryBuilder->innerJoin(
-      $subscribersTable,
-      $wpdb->prefix . 'wc_customer_lookup',
-      'customer',
-      "$subscribersTable.email = customer.email $collation"
-    )->leftJoin(
-      'customer',
-      $wpdb->prefix . 'wc_order_stats',
-      'orderStats',
-      'customer.customer_id = orderStats.customer_id AND orderStats.date_created >= :date' . $parameterSuffix . ' AND orderStats.status NOT IN ("wc-cancelled", "wc-failed")'
-    )->setParameter('date' . $parameterSuffix, $date->toDateTimeString())
+    $subQuery = $this->entityManager->getConnection()
+      ->createQueryBuilder()
+      ->from($wpdb->prefix . 'wc_customer_lookup', "customer")
+      ->select("customer.email $collation as email")
+      ->addSelect("orderStats.order_id as oder_stats_id")
+      ->leftJoin(
+        'customer',
+        $wpdb->prefix . 'wc_order_stats',
+        'orderStats',
+        'customer.customer_id = orderStats.customer_id AND orderStats.date_created >= :date' . $parameterSuffix . ' AND orderStats.status NOT IN ("wc-cancelled", "wc-failed")'
+      );
+
+    $queryBuilder->add('join', [
+      $subscribersTable => [
+        'joinType' => $this-> shouldIncludeNoneCustomerSubscribers($type, $count) ? 'left' : 'inner',
```

Comment:
In my opinion, you don't need to check which join is suitable. The inner join will be a good choice if you move the having clause to the subselect.
But be aware that you need to add a group by clause and select only emails.

I think that the final solution will have better readability.